engine/classes/Elgg/Widgets/DefaultWidgetsContainerPermissionsHandler.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Elgg\Widgets;
+
+/**
+ * Bypasses permissions for default widgets
+ *
+ * @since 4.0
+ */
+class DefaultWidgetsContainerPermissionsHandler {
+	
+	/**
+	 * Overrides permissions checks when creating widgets for logged out users.
+	 *
+	 * @param \Elgg\Hook $hook 'container_permissions_check', 'object'
+	 *
+	 * @return void|true
+	 */
+	public function __invoke(\Elgg\Hook $hook) {
+		if ($hook->getType() !== 'object' || $hook->getParam('subtype') !== 'widget') {
```

Comment:
the hook is registered for type ``object`` so why check again?

Reply:
no reason.. will change